src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/ConfirmedBlockHeadersProviderTest.java

```diff
@@ -180,9 +169,42 @@ public void testGetConfirmedBlockHeadersHSMVersion3BelowDifficultyCap() {
             difficultyCapRegTest,
             HSM_VERSION_3);
 
-        List<BlockHeader> listConfirmed = confirmedBlockHeadersProvider.getConfirmedBlockHeaders(startingPoint);
+        List<Block> confirmedBlocks = confirmedBlockHeadersProvider.getConfirmedBlocks(startingPoint);
 
         // Assert 20 elements in confirmed and 10 in potential list
-        Assert.assertEquals(30, listConfirmed.size());
+        Assert.assertEquals(30, confirmedBlocks.size());
+    }
+
+    @Test
+    public void test_getConfirmedBlocks_considerBrothersDifficultyHSMVersion3() {
```

Comment:
It would be nice to have one setup of blocks that we can test with different constructions of ConfirmedBlockHeadersProvider.

For example, we create a set of blocks with their uncles and difficulty then we test getConfirmedBlocks
- using HSM2, returns certain blocks
- using HSM3, returns other certain blocks because it considers the uncles
- high difficulty cap
- low difficulty cap

I think that would make the tests clearer and easier to understand the different cases we are covering

Reply:
Updated